@xen-orchestra/xapi/host.mjs

```diff
@@ -69,7 +70,12 @@ class Host {
     if (await this.getField('host', ref, 'enabled')) {
       await this.callAsync('host.disable', ref)
       $defer(async () => {
-        await this.callAsync('host.enable', ref)
+        await pRetry(() => this.callAsync('host.enable', ref), {
+          delay: 10e3,
+          retries: 6,
+          when: { code: 'HOST_STILL_BOOTING' },
+        })
```

Comment:
Instead of doing retries, `waitAgentRestart` won't work?
```js
...
await this.callAsync('host.disable', ref)
const prevAgentStartTime = +(await this.getField('host', ref, 'other_config')).agent_start_time
$defer(async () => {
  await waitAgentRestart(this, ref, prevAgentStartTime)
  await this.callAsync('host.enable', ref)
  ...
})

Reply:
This piece of code is in a deferred block, which means it executes after the end of the function which already calls `waitAgentRestart()`.

It means it's a good start but not enough :confused: